### ISSUE TITLE
Execution with empty shard is failing

### ISSUE THREAD

**Issue by cdsap** [square/spoon#411]
We are providing a tool for the developers of the team to execute suites of tests in our lab of devices with Spoon(always we use shards). 
Sometimes developers select suites with less than 10 test and the shard of the instrumentation includes some indexes with 0 tests.
The problem for us is that the build is marked as failed becuase `SpoonRunner.Builder().build().run()` returns false. When the `SpoonRunner` is parsing the overall success for each device we found:

      if (!result.getExceptions().isEmpty() || result.getTestResults().isEmpty()) {
              return false; // Top-level exception present, or no tests were run.
      }

So the execution fails if no test were run.

An example of Spoon output would be:
<img width="1001" alt="screen shot 2560-01-18 at 6 39 34 pm" src="https://cloud.githubusercontent.com/assets/475733/22092397/e662c852-de2f-11e6-97cb-a8d08ef20433.png">

Any ideas?

**Comment by edenman** [comment]
@LesterTheTester thoughts?  seems like what we probably want to do is use a subset of the devices if the # of tests is > # of devices?

**Comment by LesterTheTester** [comment]
Yeah, I'd agree with internally reducing the shard count in this case. It's important to preserve the behaviour where not running any tests is considered a failure.

But I'm not seeing if spoon has a concept of how many tests are going to be executed prior to the run.  If you can point me to where to get that I'm happy to make the changes
